cmd/multiclusterapp.go

```diff
@@ -679,45 +763,203 @@ func fromMultiClusterAppAnswers(answers []managementClient.Answer) map[string]st
+}
+
+func deleteMcappTargetProject(ctx *cli.Context) error {
+	c, err := GetClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	app, input, err := proceedUpdateMultiClusterAppTargetsArgs(ctx, c)
+	if err != nil {
+		return err
+	}
+	return c.ManagementClient.MultiClusterApp.ActionRemoveProjects(app, input)
+}
+
+func proceedUpdateMultiClusterAppTargetsArgs(ctx *cli.Context, c *cliclient.MasterClient) (*managementClient.MultiClusterApp, *managementClient.UpdateMultiClusterAppTargetsInput, error) {
+	if len(ctx.Args()) < 2 {
+		return nil, nil, cli.ShowSubcommandHelp(ctx)
+	}
+
+	_, app, err := searchForMcapp(c, ctx.Args().First())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	targets := ctx.Args()[1:]
+	projectIDs, err := lookupProjectIDsFromTargets(c, targets)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	answers := make(map[string]string)
+	err = processAnswers(ctx, c, nil, answers, false)
+	if err != nil {
+		return nil, nil, err
+	}
+	mcaAnswers, err := toMultiClusterAppAnswers(c, answers)
+	if err != nil {
+		return nil, nil, err
+	}
+	input := &managementClient.UpdateMultiClusterAppTargetsInput{
+		Projects: projectIDs,
+		Answers:  mcaAnswers,
+	}
+	return app, input, nil
+}
+
+func addMcappMember(ctx *cli.Context) error {
+	if len(ctx.Args()) < 3 {
 		return cli.ShowSubcommandHelp(ctx)
 	}
 
+	appName := ctx.Args().First()
+	accessType := ctx.Args().Get(1)
+	memberNames := ctx.Args()[2:]
+
+	if !slice.ContainsString(memberAccessTypes, accessType) {
+		return fmt.Errorf("invalid access type %q, supported values are \"owner\",\"member\" and \"read-only\"", accessType)
+	}
+
 	c, err := GetClient(ctx)
 	if err != nil {
 		return err
 	}
 
-	err = outputMultiClusterAppRevisions(ctx, c)
+	_, app, err := searchForMcapp(c, appName)
 	if err != nil {
 		return err
 	}
 
-	fmt.Println()
+	for _, name := range memberNames {
+		member, err := searchForMember(ctx, c, name)
+		if err != nil {
+			return err
+		}
+
+		toAddMember := managementClient.Member{
+			AccessType: accessType,
+		}
+		if member.PrincipalType == "user" {
+			toAddMember.UserPrincipalID = member.ID
+		} else {
+			toAddMember.GroupPrincipalID = member.ID
+		}
 
-	err = outputMultiClusterAppVersions(ctx, c)
+		app.Members = append(app.Members, toAddMember)
+	}
+
+	_, err = c.ManagementClient.MultiClusterApp.Update(app, app)
+	return err
+}
+
+func deleteMcappMember(ctx *cli.Context) error {
+	if len(ctx.Args()) < 2 {
+		return cli.ShowSubcommandHelp(ctx)
+	}
+
+	appName := ctx.Args().First()
+	memberNames := ctx.Args()[1:]
+
+	c, err := GetClient(ctx)
 	if err != nil {
 		return err
 	}
-	return nil
+
+	_, app, err := searchForMcapp(c, appName)
+	if err != nil {
+		return err
+	}
+
+	for _, name := range memberNames {
+		member, err := searchForMember(ctx, c, name)
+		if err != nil {
+			return err
+		}
+
+		memberID := member.ID
+		toKeepMembers := make([]managementClient.Member, 0, len(app.Members))
```

Comment:
This isn't needed. `var toKeepMembers []managementClient.Member` is fine for this.